.github/workflows/ci.yml

```diff
@@ -321,10 +324,29 @@ jobs:
         run: |
           ./build_tools/scripts/check_cuda.sh
           ./build_tools/scripts/check_vulkan.sh
-      - name: "Downloading build dir archive"
-        run: gcloud storage cp "${BUILD_DIR_GCS_ARTIFACT}" "${BUILD_DIR_ARCHIVE}"
-      - name: "Extracting build dir archive"
-        run: tar -xf "${BUILD_DIR_ARCHIVE}"
+      - name: "Downloading install dir archive"
+        run: gcloud storage cp "${INSTALL_DIR_GCS_ARTIFACT}" "${INSTALL_DIR_ARCHIVE}"
+      - name: "Extracting install dir archive"
+        run: tar -xf "${INSTALL_DIR_ARCHIVE}"
+      - name: "Configuring runtime build"
+        run: |
+          ./build_tools/github_actions/docker_run.sh \
+              gcr.io/iree-oss/nvidia@sha256:892fefbdf90c93b407303adadfa87f22c0f1e84b7e819e69643c78fc5927c2ba \
+              cmake -B ${BUILD_DIR} -S . -G Ninja \
+                  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+                  -DIREE_BUILD_COMPILER=OFF \
+                  -DIREE_BUILD_INTEGRATION_TESTS=ON \
+                  -DIREE_HOST_BIN_DIR=${INSTALL_DIR}/bin
```

Comment:
> are we even running tests?

* Baseline: https://github.com/openxla/iree/actions/runs/7647953292/job/20840561210#step:7:1182
* This PR: https://github.com/openxla/iree/actions/runs/7659217597/job/20874202528?pr=16216#step:10:1157

420 tests --> 407 tests

These tests were dropped by switching from a full compiler source build to a runtime build using install artifacts:

```
iree/samples/custom_dispatch/vulkan/shaders/example_inline.mlir.test
iree/samples/custom_dispatch/vulkan/shaders/example.mlir.test
iree/tests/e2e/regression/fill_i64.mlir.test
iree/tests/e2e/regression/force_single_dispatch.mlir.test
iree/tests/e2e/regression/globals.mlir.test
iree/tests/e2e/regression/libm_linking.mlir.test
iree/tests/e2e/regression/scalar.mlir.test
iree/tests/e2e/regression/trace_dispatch_tensors.mlir.test
iree/tests/e2e/regression/unused_args.mlir.test
iree/tests/e2e/stablehlo_models/mnist_train_test/mnist_train_test_cuda
iree/tests/e2e/stablehlo_models/mnist_train_test/mnist_train_test_vulkan
```


* The `samples/custom_dispatch/vulkan/shaders` tests use [this `CMakeLists.txt`](https://github.com/openxla/iree/blob/main/samples/custom_dispatch/vulkan/shaders/CMakeLists.txt), which could be updated to use `IREE_HOST_BIN_DIR`... if we also support `iree_lit_test_suite` (and bundle lit, FileCheck, etc. in main-dist)
* The `e2e/regression/` tests are coming from this lit test suite: https://github.com/openxla/iree/blob/eef24a93ed6235a12828aef6ab94f3ae629f1aab/tests/e2e/regression/CMakeLists.txt#L13-L33
* `mnist_train_test` is written in Python and could use a package install instead of this filtering: https://github.com/openxla/iree/blob/eef24a93ed6235a12828aef6ab94f3ae629f1aab/tests/e2e/stablehlo_models/mnist_train_test/CMakeLists.txt#L9-L14